astrodb_utils/publications.py

```diff
@@ -267,117 +250,52 @@ def ingest_publication(
         logger.error("Publication, DOI, or Bibcode is required input")
         return
 
-    if not ignore_ads:
-        ads_token = check_ads_token()
-
-        if not ads_token:
-            logger.warning(
-                "An ADS_TOKEN environment variable is not set.\n"
-                "setting ignore_ads=True.")
-            ignore_ads = True
-
-            if (not reference and (not doi or not bibcode)):
-                logger.error(
-                    "An ADS_TOKEN environment variable must be set"
-                    "in order to auto-populate the fields.\n"
-                    "Without an ADS_TOKEN, name and bibcode or DOI must be set explicity."
-                )
-                return
-   
+    if reference and not bibcode and not doi:
+        name_add = reference
+        using = f"ref: {reference} user input. No bibcode or DOI provided."
+
+    if ignore_ads is False and check_ads_token() is False:
```

Comment:
Here, I would have done the opposite since these are booleans: `if not ignore_ads and not check_ads_token()`